model_compression_toolkit/core/common/substitutions/apply_substitutions.py

```diff
@@ -13,14 +13,12 @@
 # limitations under the License.
 # ==============================================================================
 
-import copy
-
 from typing import List
 
 from model_compression_toolkit.core import common
 
 
-def substitute(graph_to_substitute: common.Graph,
+def substitute(graph: common.Graph,
```

Comment:
This one seems a bit dangerous. How can we verify that there in no one substitution that is counting on the fact that the graph is being copied? I'm not sure that our tests can ensure this.